configure.ac

```diff
@@ -483,6 +485,41 @@ AC_CHECK_HEADERS([sys/types.h signal.h stddef.h], [],
 # optional:
 AC_CHECK_HEADERS([sys/time.h locale.h fcntl.h dlfcn.h sys/wait.h sys/sysmacros.h])
 
+# Check for JNI
+AC_ARG_ENABLE([java],
+              [AC_HELP_STRING([--disable-java],
+                              [disable Java Interoperability])])
+AS_IF([test X$enable_java != Xno], [
+   AX_JNI_INCLUDE_DIR
+   for JNI_INCLUDE_DIR in $JNI_INCLUDE_DIRS
+	do
+		CPPFLAGS="$CPPFLAGS -I$JNI_INCLUDE_DIR"
+	done
+	AC_DEFINE([WITH_JNI], [1])
```

Comment:
We may also need to check there was no failure in `AX_JNI_INCLUDE_DIR` and then skip in such a case.